docs/source/dev/yaml_spec.rst

```diff
@@ -0,0 +1,441 @@
+YAML Data File Specification
+============================
+
+This file constitutes the official specification for the YAML data files used
+to store the data for an :term:`instrument`.
+
+Spec
+----
+
+The data must be stored in a YAML format file with the following structure:
+
+.. parsed-code-block:: yaml
+    :emphasize-lines: 1,2,3,5,8,9,10,15,17
+
+    :ref:`name<spec-name>`: :iref:target:`"instrument_name"<spec-name-targ>`
+    :ref:`default_version<spec-default-version>`: :iref:target:`"version1"<spec-default-version-targ>`
+    :ref:`version<spec-version>`\ :iref:target:`:<spec-version-targ>`
+        version1:
+            :ref:`default_model<spec-default-model>`: :iref:target:`"model1"<spec-default-model-targ>`
+            :ref:`models<spec-models>`\ :iref:target:`:<spec-models-targ>`
+                model1: "model1_v1"
+                model1_v1:
+                    :ref:`function<spec-function>`: :iref:target:`"model_function_name"<spec-function-targ>`
+                    :ref:`citation<spec-citation>`: ["citation1", :iref:target:`"citation2"<spec-citation-targ>`]
+                    :ref:`parameters<spec-parameters>`\ :iref:target:`:<spec-parameters-targ>`
+                        parameter1: "value1"
+                        parameter2: 2
+                        parameter3: 3.14
+                        parameter4: [1, 2, 3, 4]
+                    :ref:`configurations<spec-configurations>`\ :iref:target:`:<spec-configurations-targ>`
+                        configuration1:
+                            :ref:`default_option<spec-default-option>`: :iref:target:`"option1"<spec-default-option-targ>`
+                            option1:
+                                parameter5: "value2"
+                                parameter6: 56.1687
```

Comment:
A potential way to make the validation simpler would be by changing the YAML spec by adding an additional level of indentation via an `options` dict:

```
                    :ref:`configurations<spec-configurations>`\ :iref:target:`:<spec-configurations-targ>`
                        configuration1:
                            :ref:`default_option<spec-default-option>`: :iref:target:`"option1"<spec-default-option-targ>`
                            options:
                                option1:
                                    parameter5: "value2"
                                    parameter6: 56.1687
```